client/src/utils/testUtils.tsx

```diff
@@ -0,0 +1,27 @@
+import type { RenderOptions } from '@testing-library/react';
+import type { FC, ReactElement, ReactNode } from 'react';
+
+import { QueryClientProvider } from '@tanstack/react-query';
+import { render } from '@testing-library/react';
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { ToastContainer } from 'react-toastify';
+
+import queryClient from '@/queryClient';
+
+export const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
+  return (
+    <QueryClientProvider client={queryClient}>
+      <MemoryRouter>{children}</MemoryRouter>
+      <ToastContainer />
+    </QueryClientProvider>
+  );
+};
```

Comment:
각 테스트는 독립적이어야하는데 같은 queryClient 인스턴스를 사용하면 테스트를 병렬로 실행할 때 문제가 생길 수 있어 보입니다. provider를 만들 때마다 새로운 쿼리 클라이언트 인스턴스를 생성해서 내려주는 건 어떤가요?

Reply:
각 테스트마다 렌더링을 하고 있어서 괜찮지 않을까요? 

저도 테스트코드는 많이 작성해보지 않아서, 뒷부분 말씀하신건 이해를 못했는데, 예시 코드를 작성해주실 수 있을까요.

Reply:
```typescript
import { QueryClient } from '@tanstack/react-query'

export const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  const queryClient = new QueryClient();

  return (
    <QueryClientProvider client={queryClient}>
      <MemoryRouter>{children}</MemoryRouter>
      <ToastContainer />
    </QueryClientProvider>
  );
};
```

Reply:
하나의 테스트는 독립적이어야하므로 병렬로 실행되는 테스트가 동일한 객체 인스턴스를 사용하면 안된다는 생각에 말씀드렸습니다.

Reply:
> Whenever you use React Query, you need a QueryClientProvider and give it a queryClient - a vessel which holds the QueryCache. The cache will in turn hold the data of your queries.
> I prefer to give each test its own QueryClientProvider and create a new QueryClient for each test. That way, tests are completely isolated from each other. A different approach might be to clear the cache after each test, but I like to keep shared state between tests as minimal as possible. Otherwise, you might get unexpected and flaky results if you run your tests in parallel.

https://tkdodo.eu/blog/testing-react-query#queryclientprovider

아니면 setUpTest.ts에서 개별 테스트가 끝날 때마다 캐시를 clear해주는 방법도 있겠습니다.
```typescript
import queryClient from '@/queryClient';

afterEach(() => queryClient.clear());
```

Reply:
좋네요. 이해했습니다.

Reply:
How to clear queryCache after every Jest test run in version 3?
https://github.com/TanStack/query/discussions/1441
비슷한 논의인 거 같아서 가져와봤습니다 자세히는 저도 안 읽어봄